ush/jediinc2fv3.py

```diff
@@ -22,57 +20,174 @@
+    # Some global attributes
+    ncout.source = 'jediinc2fv3.py'
+    ncout.comment = 'Increment produced by FV3-JEDI and modified for use by FV3 read'
+
+    # Create all the dummy vertical coordinate variables
+    nlevs = len(ncin.dimensions['lev'])
+    nlats = len(ncin.dimensions['lat'])
+    nlons = len(ncin.dimensions['lon'])
+    pfull = range(1, nlevs+1)
+    phalf = range(1, nlevs+2)
+    levvar = ncout.createVariable('lev', 'f4', ('lev'))
+    levvar[:] = pfull
+    pfullvar = ncout.createVariable('pfull', 'f4', ('lev'))
+    pfullvar[:] = pfull
+    ilevvar = ncout.createVariable('ilev', 'f4', ('ilev'))
+    ilevvar[:] = phalf
+    hyaivar = ncout.createVariable('hyai', 'f4', ('ilev'))
+    hyaivar[:] = phalf
+    hybivar = ncout.createVariable('hybi', 'f4', ('ilev'))
+    hybivar[:] = phalf
+
+    # Rename and change dimensionality of fields
+    for name, variable in ncin.variables.items():
+        if len(variable.dimensions) in [3, 4]:
+            dimsout = variable.dimensions[1:]
+            dimsout_inc = dimsout
+        else:
+            dimsout = variable.dimensions
+
+        if name in vardict:
+            # The temperature increment is needed to compute the delz
+            # increment.  Either T or t may be used in jedi increment
+            # files.  Set tinc to the appropriate case.
+            if name in ['T', 't']:
+                tinc = name
+
+            x = ncout.createVariable(vardict[name], 'f4', dimsout)
+            if len(variable.dimensions) in [3, 4]:
+                ncout[vardict[name]][:] = ncin[name][0, ...]
+            else:
+                ncout[vardict[name]][:] = ncin[name][:]
+
+    # Populate increment and guess fields
+    #   Note:  increment and guess fields have same shape
+    #     ps_inc is (time, lat, lon), ps_ges is {time, grid_yt, grid_xt)
+    #     t_inc is (time, lev, lat, lon), t_ges is (time, pfull, grid_yt, grid_xt)
+
+    ps_ges = ncges.variables['pressfc'][:]
+    t_ges = ncges.variables['tmp'][:]
+    q_ges = ncges.variables['spfh'][:]
+
+    ps_inc = ncin.variables['ps'][:]
+    t_inc = ncin.variables[tinc][:]
+    q_inc = ncin.variables['sphum'][:]
+
+    # Compute analysis ps
+    ps_anl = np.zeros((nlats, nlons), float)
+    ps_anl = ps_ges + ps_inc
+
+    # Set constants and compute derived constants
+    grav = 9.80665
+    airmw = 28.965
+    h2omw = 18.015
+    runiv = 8314.47
+    rdry = runiv/airmw
+    rvap = runiv/h2omw
+    cpdry = 3.5*rdry
+    fv = (rvap/rdry)-1
+    rdog = rdry/grav
+    kappa = rdry/cpdry
+    kap1 = kappa+1.0
+    kapr = 1.0/kappa
+
+    # Get ak,bk from guess file
+    nc_attrs = ncges.ncattrs()
+    ak = ncges.getncattr('ak')
+    bk = ncges.getncattr('bk')
+
+    # Compute guess and analysis interface pressures
+    prsi_ges = np.zeros((nlevs+1, nlats, nlons), float)
+    prsi_anl = np.zeros((nlevs+1, nlats, nlons), float)
+    k = 0
+    while k < nlevs+1:
```

Comment:
Change while loops to for loops?
`for k in range(0, nlevs+1):`?

Reply:
Then lines 131 and 135 can be removed

Reply:
Python range is not inclusive on the 'right' side, so `0,nlevs+1` would be 0 through nlevs.

Reply:
Thank you, @CoryMartin-NOAA , for this suggestion.   Feedback from @guillaumevernieres and you are helping me learn new and better ways to code in python.   Thanks!   

`while` loops have been replace with `for` loops.  Done at [43f8ede](https://github.com/NOAA-EMC/GDASApp/commit/43f8ede69a58ab45e441ab12b5c717ce017f09da)